tpp-app/src/services/SettingsService.js

```diff
@@ -0,0 +1,282 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { TRACK_SYMPTOMS, REMINDERS} from './utils/constantscs'
+
+    /**
+     * Clears all of the user's account data
+     * @returns a promise resolving when the clear operation is complete
+     */
+
+    export const DELETEAccountData = async () => new Promise( async (resolve, reject) => {
+        try {
+            await AsyncStorage.clear().then(() => {
+                console.log("Deleted user account data");
+                resolve(); 
+            })
+            
+        } catch (e) {
+            console.log(`DELETEAccountData error: ${JSON.stringify(e)}`);
+            reject();
+        }
+    })
+
+    /**
+     * Returns an array of a boolean representing if a symptom is currently tracked
+     * @returns a promise of an array containing booleans representing whether to track each symptom
+     */
+    export const GETAllTrackingPreferences = async () => new Promise (async (resolve, reject) => {
+        try {
+             values = await AsyncStorage.multiGet([
+                TRACK_SYMPTOMS.FLOW,
+                TRACK_SYMPTOMS.MOOD,
+                TRACK_SYMPTOMS.SLEEP,
+                TRACK_SYMPTOMS.CRAMPS,
+                TRACK_SYMPTOMS.EXERCISE
+             ]).then((values) => {
+                 console.log("Got All Tracking Preferences");
+                 resolve(values);
+             }).catch((e) => {
+                console.log(JSON.stringify(e));
+                reject('Unable to getItem');
+             });
+        } catch (e) {
+            console.log(`GETAllTrackingPreferences error: ${JSON.stringify(e)}`)
+            reject();
+        }
+    })
+
+
+    /**
+     * Posts any changes in symptom tracking preferences 
+     * @param {boolean} flow representing whether to track flow
+     * @param {boolean} mood representing whether to track mood 
+     * @param {boolean} sleep representing whether to track sleep 
+     * @param {boolean} cramps representing whether to track cramps
+     * @param {boolean} exercise representing whether to track exercise
+     * @returns a promise resolving when the post operation is complete
+     */
+    export const POSTUpdatePreferences = async (flow, mood, sleep, cramps, exercise) => new Promise (async (resolve, reject) => {
+        try {
```

Comment:
recommend adding an `.any()` check here to make sure that at least one of the symptoms would get tracked but it should be fine if you just check for this on the frontend so feel free to not do it here :)

Reply:
+1 would suggest adding here, as if its somehow POSTed to asyncstorage that nothing's being tracked we will have issues as other places we assume at least one preference is set to TRUE